### DISCUSSION TITLE
UID on openshift cluster vs podman

### DISCUSSION THREAD

**Discussion by imperialguy** [containers/podman#16013] [General]
I have a container that runs with uid 1001. 

In order to properly map the UIDs in podman, I do this:

```
podman run -d \
	--name mycontainer \
	--uidmap 1001:0:1 \
	--gidmap 1001:0:1 \
	--uidmap 0:1:1001 \
	--gidmap 0:1:1001 \
        myimage:mytag
```
This works fine.

But, if I were to run this on an openshift cluster, with the same UID it fails because the default UID range for any namespace in opennshift is 1000600000/10000. So the following spec with fail in openshift cluster:

```
spec:
      securityContext:
        runAsUser: 1001
        runAsGroup: 1001
```

In other words, openshift starts uid at 1000600000. So, it cannot understand the UID 1001.

There's two options I have:

1. Change the default uid range in openshift from 1000600000/10000 to 1000/10000

This works, but it requires me to manually edit the namespace, and changing the default uid range just doesn't seem right. So, this means I am comprising openshift here.

2. Change the uid inside the container to something like (say) 1000600001 or beyond.

I tried doing this but, podman failed to map this on my local machine. So, the following will fail in podman:

```
podman run -d \
	--name mycontainer \
	--uidmap 1000600001:0:1 \
	--gidmap 1000600001:0:1 \
	--uidmap 0:1:1000600001 \
	--gidmap 0:1:1000600001 \
        myimage:mytag
```

How can I make podman to work with the UID 1000600001? Because I don't want to change the default UID range in openshift cluster.

**Comment by mheon** [comment]
@giuseppe PTAL
